Use bootstrap classes

xf- opened the pull request:
avoid overwriting some bootstrap stuff

Bootstrap uses min-width and not max because larger screen have usually more high-end PC and not a small smartphone. d-none and d-size-display overwrites it ad min-width.

```css
@media (min-width:768px){
```

Adjusted the max mediaquery, because it would match with min-width 768 from bootstrap on taht exact pixel.


## Scollbars
Shrink you current git state to 780px or mobile size and you will get scrollbars because 160+col-md-10 has more width than 100% of your current window.

## Table
![image](https://user-images.githubusercontent.com/1295633/75566664-5cdb9780-5a50-11ea-9b68-5d841fbc7faf.png)
(Scrollbar is OS related - can be styled in last versions for windows and Linux. Not visible on some touch devices, but works with normal gestures)

varjolintu commented:
This messes up the table view. Now the width is always 100% of the page, plus the elements inside are no longer centered. You can clearly see that in the picture. It's preferred that the width is not 100% unless the screen is too small to contain everything.

![Screen Shot 2020-02-29 at 9 57 10](https://user-images.githubusercontent.com/24570482/75603719-e2575a00-5ad9-11ea-9100-d25e0e9005fb.png)

xf- commented:
On current state, the UI has some min-width even with sidebar menu you can get a horizontal scrollbar
![image](https://user-images.githubusercontent.com/1295633/75608983-f2386380-5b04-11ea-8dcc-777d031ed6d5.png)

I removed the min with and also tried to get rid of multiline table headers. Maybe translations in some languages are even more lines. I also removed the break-all e.g.
![image](https://user-images.githubusercontent.com/1295633/75608997-23b12f00-5b05-11ea-8854-97167df6dacd.png)

For the scrollbar i made the tables responsive. Easier to navigate.

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
     font-size: 14px;
     padding-top: 20px;
     padding-bottom: 60px;
+    max-width: 1440px;
```
added a max-width like we wrote in irc

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
 body {
-    background-color: var(--background-color);
```
removed all duplicate/overwritten values to media query

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
-    text-decoration: none;
-}
-
-table td {
```
made the tables responsive

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
 }
 
 input[type=range] {
-    width: 20%;
```
20% depends a lot on screen size

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
 footer {
     bottom: 1rem;
-    color: #cbcfcb;
-    font-size: .75rem;
```
using classes from bootstrap css

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
 
-td.identifier, td.key, td.last-used, td.created {
-    white-space: nowrap;
+#tab-site-preferences td:nth-of-type(2) {
```
table width in site preferences

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -1,114 +1,68 @@
-.card-body {
-    background-color: var(--background-color);
-    color: var(--text-color);
+    border-radius: 0;
```
bootstrap css again

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -128,114 +81,40 @@ button.collapsed {
-        display: none !important;
-    }
-
+@media (min-width: 768px) {
```
use min-width -> bootstrap works the same way and get rid of most selectors and use existing classes

xf- commented on keepassxc-browser/options/options.css:
```diff
@@ -251,13 +130,9 @@ button.collapsed {
         color: var(--text-color);
     }
 
-    .navbar-default {
```
moved style to upper section. no var

xf- commented on keepassxc-browser/options/options.html:
```diff
@@ -20,41 +20,41 @@
-            <h6 class="sidebar-heading d-flex px-3 mt-4 mb-1">KeePassXC-Browser</h6>
-            <ul class="nav flex-column mb-2">
+          <div class="collapse d-none d-md-block" id="sidebar">
+            <h1 class="text-uppercase text-white-50 d-none d-md-flex px-3 pb-4 mt-4 mb-1 ">KeePassXC-Browser</h1>
```
no h1 on the page before

xf- commented on keepassxc-browser/options/options.html:
```diff
@@ -271,8 +273,8 @@ <h2 class="tab-header" data-i18n="optionsGeneralSettingsTab"></h2>
-              <div class="card shadow">
-                <div class="card-header">
+              <div class="card my-4 shadow">
+                <div class="card-header h6">
```
h6 or h6 class is 1rem and fold - same as applied style in css before

xf- commented on keepassxc-browser/options/options.js:
```diff
@@ -31,7 +31,7 @@ options.initMenu = function() {
         $(this).parent('li').addClass('active');
         $('div.tab').hide();
-        $('div.tab#tab-' + $(this).attr('href').substring(1)).fadeIn();
+        $('div.tab#tab-' + $(this).attr('href').substring(1)).removeClass('d-none').fadeIn();
```
using d-none instead of a extra display none definition

xf- commented on keepassxc-browser/options/options.js:
```diff
@@ -302,7 +302,7 @@ options.initConnectedDatabases = function() {
 
     const trClone = $('#tab-connected-databases table tr.clone:first').clone(true);
-    trClone.removeClass('clone');
+    trClone.removeClass('clone d-none');
```
using d-none instead of a extra display none definition

xf- commented on keepassxc-browser/options/options.html:
```diff
@@ -599,8 +607,8 @@ <h2 class="tab-header" data-i18n="optionsAboutTab"></h2>
 
-              <div class="card shadow">
-                <div class="card-header">
+              <div class="card my-4 shadow">
```
add 1.5rem space - i hope that is OK. looks the same as the space to the sidebar.

varjolintu commented:
There are still issues:
- When resizing the window to under 992px the sidebar width changes to a bigger size
- The table content is still not vertically centered (see screenshot)
- Extra whitespace around the checkboxes (doesn't look very nice)

![Screen Shot 2020-03-01 at 14 58 09](https://user-images.githubusercontent.com/24570482/75626145-8a9a1b00-5bcd-11ea-9f11-ca2993ea4e78.png)

xf- commented:
>     * When resizing the window to under 992px the sidebar width changes to a bigger size (from around 164px to 247 px etc.)

You used a fix min width for the menu. currently it uses bootstrap responsive design > lg the menu has 2 cols and and md to lg 3 cols for the 12 col grid. With md 2 cols the items in menu using 2 lines and it is enough space on screen for the menu. lg with 3 cols is no problem, but seems more lost again.

>     * The table content is still not vertically centered (see screenshot)
Never changed something - is this way int he develop (at least on my system)

> 
>     * Extra whitespace around the checkboxes (doesn't look very nice)
OK - i change it back to align right
> 
> 
> ![Screen Shot 2020-03-01 at 14 58 09](https://user-images.githubusercontent.com/24570482/75626145-8a9a1b00-5bcd-11ea-9f11-ca2993ea4e78.png)

In general i would prefer cards design over the table in the future, because everything is together on screen and you don't need to follow a line to set something like username.
https://getbootstrap.com/docs/4.4/components/card/#card-decks

xf- commented:
@varjolintu 
![image](https://user-images.githubusercontent.com/1295633/75635402-1172e600-5c16-11ea-9596-c17eb95cf01b.png)
Always 3 cols - menu gets really large and 2 cols is not enough on md (> 768).

xf- commented:
@varjolintu about sidebar - the current behavior or more width >= lg. This kind of behavior is used often on websites. Normal user won't resize the window.

Everything else OK?

varjolintu commented:
@xf- It's better now.

A few small issues:

That combo box is way too wide when compared to the initial one.
![Screen Shot 2020-03-08 at 20 38 48](https://user-images.githubusercontent.com/24570482/76168985-f9064c80-617c-11ea-9cfa-98f03bca0ac3.png)

Is it possible to keep the spacing? See the picture below for the current layout. And I'm still not sure why the _Username-only Detection_ cell must be so wide. The current layout has a smaller one, and it looks better IMO.
![Screen Shot 2020-03-08 at 20 38 58](https://user-images.githubusercontent.com/24570482/76169001-0b808600-617d-11ea-808f-3dbfa5bf6fd7.png)

![Screen Shot 2020-03-08 at 20 39 27](https://user-images.githubusercontent.com/24570482/76169003-16d3b180-617d-11ea-8cd8-2fd6a8b7c1b3.png)

xf- commented:
The UI is translated and depending on translation/language and line break behavior column header  can get really ugly. a fix with should keep the title in 2 lines. CZ maybe around 4 lines and other languages even more. Depending on the translation length.

About the select (combobox) - i suggest instead of fix % (like before - i used a bootstrap class) we use 100% < md and md above 3 cols. Currently it isn't really responsive on small screens.

varjolintu commented:
If we can, we should keep the column header width as long as the string is (like it's currently). There's no need for any extra space.

xf- commented:
As discussed some responsive changes

- Alerts, inputs with col widths instead of static %
- Table width more responsive. use the width if available
- Using single selector per line (more overview - i can put all in a line at the end again)

varjolintu commented:
The selection is still too large. Didn't you just fixed this?
![Screen Shot 2020-03-12 at 9 44 53](https://user-images.githubusercontent.com/24570482/76498759-2d605e00-6446-11ea-9b1d-a2804e233231.png)

Now there's some extra space in the delete cell too:
![Screen Shot 2020-03-12 at 9 45 38](https://user-images.githubusercontent.com/24570482/76498810-4a952c80-6446-11ea-99eb-b4fd15b7db1d.png)

It's even worse in the Custom Login table:
![Screen Shot 2020-03-12 at 9 46 25](https://user-images.githubusercontent.com/24570482/76498848-5a147580-6446-11ea-875d-5f9ad88a41f6.png)
